spikeinterface/extractors/nwbextractors.py

```diff
@@ -47,18 +62,10 @@ def __init__(self, file_path: PathType, electrical_series_name: str = None):
         """
         assert HAVE_NWB, self.installation_mesg
         self._file_path = str(file_path)
-        with NWBHDF5IO(self._file_path, 'r') as io:
+        self._electrical_series_name = electrical_series_name
+        with NWBHDF5IO(self._file_path, mode='r',load_namespaces=True) as io:
             nwbfile = io.read()
-            if electrical_series_name is not None:
-                electrical_series_name = electrical_series_name
-            else:
-                a_names = list(nwbfile.acquisition)
-                if len(a_names) > 1:
-                    raise ValueError("More than one acquisition found! You must specify 'electrical_series_name'.")
-                if len(a_names) == 0:
-                    raise ValueError("No acquisitions found in the .nwb file.")
-                electrical_series_name = a_names[0]
-            es = nwbfile.acquisition[electrical_series_name]
+            es = get_electrical_series(nwbfile,self._electrical_series_name)
```

Comment:
I don't anderstand the logic here.
Why don't you assign back `es` to `self._electrical_series_name`

Reply:
es is a container that is only accessible when the file is open, which closes at the end of __init__. Since it cannot be used by other methods, it does not make sense to store it as a attribute. 
Also, `self._electrical_series_name` is a str type storing only the actual name for hte electricalseries to be retrieved from the nwb file each  time its opened